src/Hedgehog/Linq/Property.fs

```diff
@@ -6,46 +6,53 @@ open System
 open System.Runtime.CompilerServices
 open Hedgehog
 
-[<Extension>]
-[<AbstractClass; Sealed>]
-type Property private () =
+type Property = private Property of Property<unit> with
 
-    static member Failure : Property<unit> =
+    static member Failure : Property =
         Property.failure
+        |> Property
 
-    static member Discard : Property<unit> =
+    static member Discard : Property =
         Property.discard
+        |> Property
 
     static member Success (value : 'T) : Property<'T> =
         Property.success value
 
-    static member FromBool (value : bool) : Property<unit> =
+    static member FromBool (value : bool) : Property =
         Property.ofBool value
+        |> Property
 
     static member FromGen (gen : Gen<Journal * Result<'T>>) : Property<'T> =
         Property.ofGen gen
 
     static member FromResult (result : Result<'T>) : Property<'T> =
         Property.ofResult result
 
-    static member FromThrowing (throwingFunc : Action<'T>, arg : 'T) : Property<unit> =
+    static member FromThrowing (throwingFunc : Action<'T>, arg : 'T) : Property =
         Property.ofThrowing throwingFunc.Invoke arg
+        |> Property
 
     static member Delay (f : Func<Property<'T>>) : Property<'T> =
         Property.delay f.Invoke
 
     static member Using (resource : 'T, action : Func<'T, Property<'TResult>>) : Property<'TResult> =
         Property.using resource action.Invoke
 
-    static member CounterExample (message : Func<string>) : Property<unit> =
+    static member CounterExample (message : Func<string>) : Property =
         Property.counterexample(message.Invoke)
+        |> Property
 
     static member ForAll (gen : Gen<'T>, k : Func<'T, Property<'TResult>>) : Property<'TResult> =
         Property.forAll gen k.Invoke
 
     static member ForAll (gen : Gen<'T>) : Property<'T> =
         Property.forAll' gen
 
+[<Extension>]
+[<AbstractClass; Sealed>]
+type PropertyExtensions private () =
```

Comment:
The only disadvantage with `PropertyExtensions` is that the rest of the classes (Range, Gen) in Hedgehog.Linq don't have the *Extensions* suffix — (fortunately).

.NET extensions are sometimes considered as a [*feature envy*](https://twitter.com/ploeh/status/253426390718881792). If we'd want to be consistent I'd suggest we consider naming this `PropertyEnvy`, and then `GenEnvy`, `RangeEnvy`.

These can be ugly names for some people going through the codebase, so perhaps it's worth trying a trick like this
https://github.com/hedgehogqa/fsharp-hedgehog/blob/f0ea8a1a35eefd7ae0f5734fdef5eb1d3b75064e/src/Hedgehog/Seed.fs#L37

The rest of the PR looks good, and I don't want this comment of mine to be a blocker issue. If @adam-becker or @TysonMN have a different opinion we can consider that as well.

Reply:
I think the name `PropertyExtension` is fine.  This is very standard in C#.

For consistency, we can also move the `Gen<_>` extensions and `Range<_>` extension to `GenExtensions` and `RangeExtensions`.

Reply:
OK, sounds like a plan.

Reply:
So are we ready to merge or do we want to rename also the types inside /Linq/Gen.fs and /Ling/Range.fs so that they end with *Extensions*? (I wouldn't yet rename the files, because we might end up adding more stuff to those as it happened here.)

Reply:
Making that change in another PR is fine with me.

And I wasn't thinking just a rename.  I was thinking that the extensions in `Gen` move to `GenExtensions` and the non-extensions stay on `Gen`.

Reply:
Thanks! Let's wait for @adam-becker (just in case he plans to do that in this PR) before we can merge.

Reply:
Thanks for waiting! I went ahead and made this change as well.